src/main/scala/viper/gobra/frontend/info/implementation/property/Satisfiability.scala

```diff
@@ -0,0 +1,29 @@
+package viper.gobra.frontend.info.implementation.property
+
+import org.bitbucket.inkytonik.kiama.util.Messaging.error
+import viper.gobra.ast.frontend.{PIdnUse, PInterfaceType, PNamedOperand, PType, PTypeElement}
+import viper.gobra.frontend.info.base.Type._
+import viper.gobra.frontend.info.implementation.TypeInfoImpl
+import viper.gobra.frontend.info.implementation.resolution.TypeSet
+
+trait Satisfiability extends BaseProperty { this: TypeInfoImpl =>
+
+  lazy val satisfies: Property[(Type, PInterfaceType)] = createProperty {
+    case (t, PInterfaceType(Vector(PTypeElement(Vector(PNamedOperand(PIdnUse("comparable"))))), methSpecs, _)) =>
+      comparableType.result(t) and goImplements.result(t, PInterfaceType(Vector(), methSpecs, Vector()))
+    case (t, constraintType) =>
+      goImplements.result(t, constraintType)
+  }
+
+  private lazy val goImplements: Property[(Type, PInterfaceType)] = createProperty {
```

Comment:
the name is very similar to `syntaxImplements`. Please change the name.

Reply:
I named it like this, because this actually reflects the definition of `implements` from Go. What do you think of renaming `syntaxImplements` to `memberSetImplements` or `implementsMemberSet`?

Reply:
changing `syntaxImplements` to `implementsMemberSet` is ok.